tests/std/tests/P0896R4_ranges_alg_replace_copy/test.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+using P = pair<int, int>;
+
+// Validate that replace_copy_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::replace_copy_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::replace_copy(borrowed<false>{}, static_cast<int*>(nullptr), 42, 5)),
+    ranges::replace_copy_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::replace_copy(borrowed<true>{}, static_cast<int*>(nullptr), 42, 5)),
+    ranges::replace_copy_result<int*, int*>>);
+
+struct instantiator {
+    static constexpr P input[5]    = {{0, 99}, {1, 47}, {2, 99}, {3, 47}, {4, 99}};
+    static constexpr P expected[5] = {{0, 99}, {47, 1}, {2, 99}, {47, 1}, {4, 99}};
+
+    template <ranges::input_range Read, indirectly_writable<ranges::range_reference_t<Read>> Write>
+    static constexpr void call() {
+        using ranges::replace_copy, ranges::replace_copy_result, ranges::iterator_t;
+        { // Validate iterator + sentinel overload
+            P output[5] = {{-1, -1}, {-1, -1}, {-1, -1}, {-1, -1}, {-1, -1}};
+            Read wrapped_input{input};
+
+            auto result =
+                replace_copy(wrapped_input.begin(), wrapped_input.end(), Write{output}, 47, P{47, 1}, get_second);

```

Comment:
The compiler error is:

```
C:\agent\_work\1\s\tests\std\tests\P0896R4_ranges_alg_replace_copy\test.cpp(35,17): error: no matching function for call to object of type 'const std::ranges::_Replace_copy_fn'
                replace_copy(wrapped_input.begin(), wrapped_input.end(), Write{output}, 47, P{47, 1}, get_second);
                ^~~~~~~~~~~~
C:/agent/_work/1/s/tests/std/include\range_algorithm_support.hpp(603,32): note: in instantiation of function template specialization 'instantiator::call<test::range<std::input_iterator_tag, const std::pair<int, int>, test::Sized::no, test::CanDifference::no, test::Common::no, test::CanCompare::no, test::ProxyRef::no>, test::iterator<std::input_iterator_tag, std::pair<int, int>, test::CanDifference::no, test::CanCompare::no, test::ProxyRef::no, test::IsWrapped::yes> >' requested here
        Continuation::template call<Args...,
                               ^
C:/agent/_work/1/s/tests/std/include\range_algorithm_support.hpp(770,32): note: in instantiation of function template specialization 'with_writable_iterators<instantiator, std::pair<int, int> >::call<test::range<std::input_iterator_tag, const std::pair<int, int>, test::Sized::no, test::CanDifference::no, test::Common::no, test::CanCompare::no, test::ProxyRef::no> >' requested here
        Continuation::template call<Args...,
                               ^
C:/agent/_work/1/s/tests/std/include\range_algorithm_support.hpp(945,83): note: in instantiation of function template specialization 'with_input_ranges<with_writable_iterators<instantiator, std::pair<int, int> >, const std::pair<int, int> >::call<>' requested here
    with_input_ranges<with_writable_iterators<Instantiator, Element2>, Element1>::call();
                                                                                  ^
C:\agent\_work\1\s\tests\std\tests\P0896R4_ranges_alg_replace_copy\test.cpp(55,20): note: in instantiation of function template specialization 'test_in_write<instantiator, const std::pair<int, int>, std::pair<int, int> >' requested here
    STATIC_ASSERT((test_in_write<instantiator, P const, P>(), true));
                   ^
D:\build\x64\out\inc\algorithm(3080,50): note: candidate template ignored: constraints not satisfied [with _It = test::iterator<std::input_iterator_tag, const std::pair<int, int>, test::CanDifference::no, test::CanCompare::no, test::ProxyRef::no, test::IsWrapped::yes>, _Se = test::sentinel<const std::pair<int, int>, test::IsWrapped::yes>, _Ty1 = int, _Ty2 = std::pair<int, int>, _Out = test::iterator<std::input_iterator_tag, std::pair<int, int>, test::CanDifference::no, test::CanCompare::no, test::ProxyRef::no, test::IsWrapped::yes>, _Pj = get_nth_fn<1>]
        constexpr replace_copy_result<_It, _Out> operator()(
                                                 ^
D:\build\x64\out\inc\algorithm(3076,86): note: because 'output_iterator<test::iterator<std::input_iterator_tag, std::pair<int, int>, test::CanDifference::no, test::CanCompare::no, test::ProxyRef::no, test::IsWrapped::yes>, const std::pair<int, int> &>' evaluated to false
        template <input_iterator _It, sentinel_for<_It> _Se, class _Ty1, class _Ty2, output_iterator<const _Ty2&> _Out,
                                                                                     ^
D:\build\x64\out\inc\xutility(844,9): note: because '*__i++ = static_cast<_Ty &&>(__t)' would be invalid: indirection requires pointer operand ('void' invalid)
        *__i++ = static_cast<_Ty&&>(__t);
        ^
```

The template machinery here is almost beyond my comprehension, but I observe that (1) your test differs from the other tests currently using `test_in_write` in that its `instantiator` actually verifies the `indirectly_writable` concept, and (2) `with_writable_iterators` appears to pass `iterator<input, MEOW>` which doesn't seem correct:

https://github.com/microsoft/STL/blob/5be7d49c243979231dff35919d3a3d813d75d714/tests/std/include/range_algorithm_support.hpp#L603-L606

I'm not sure if this analysis is correct.